pyccel/ast/cwrapper.py

```diff
@@ -114,7 +114,7 @@ def __init__(self, python_func_args, python_func_kwargs, c_func_args, parse_args
             raise TypeError('Python func args should be a Variable')
         if not isinstance(python_func_kwargs, Variable):
             raise TypeError('Python func kwargs should be a Variable')
-        if not isinstance(c_func_args, list) and any(not isinstance(c, Variable) for c in c_func_args):
+        if not isinstance(c_func_args, list) and any(not isinstance(c, (Variable, FunctionAddress)) for c in c_func_args):
```

Comment:
it's should be an `or`. however we pass a `c_func_args` sympy tuple and not a list. I think I should perform a test to check if `c_func_args` is iterable insead

Reply:
It makes sense to use tuples in all situations where we don't plan to modify the list. Let me know what you find out please!

Reply:
I checked, we don't need to change the list, and I don't think we will in this context, we only use it to get informations about python arguments that will be parsed

Reply:
Then it seems that we don't need to check if `c_func_args` is a list, and the correct check here should be
```python
if any (not isinstance(c, (Variable, FunctionAddress)) for c in c_func_args)
```
or (a little easier to read)
```python
if not all(isinstance(c, (Variable, FunctionAddress)) for c in c_func_args)
```

@EmilyBourne @rmahjoubi What do you think?

Reply:
I agree, if you iterate over something that is not iterable then the error should be enough to tell what the problem is. If we'd rather keep it as it is then we should update the error message to specify a tuple of Variables rather than a list of Variables